extensions-contrib/prometheus-emitter/src/main/java/org/apache/druid/emitter/prometheus/DimensionsAndCollector.java

```diff
@@ -66,22 +70,41 @@ public double[] getHistogramBuckets()
     return histogramBuckets;
   }
 
-  public void resetLastUpdateTime()
+  /**
+   * For each unique set of labelValues, keeps track of the amount of time that has elapsed since its metric
+   * value has been updated.
+   */
+  public void resetLastUpdateTime(List<String> labelValues)
   {
-    updateTimer.restart();
+    labelValuesToStopwatch.compute(labelValues, (k, v) -> {
+      if (v != null) {
+        v.restart();
+        return v;
+      } else {
+        return Stopwatch.createStarted();
+      }
+    });
   }
 
-  public long getMillisSinceLastUpdate()
+  public ConcurrentMap<List<String>, Stopwatch> getLabelValuesToStopwatch()
   {
-    return updateTimer.millisElapsed();
+    return labelValuesToStopwatch;
   }
 
-  public boolean isExpired()
+  /**
+   * For the given labelValues, checks if the metric value has been updated within the configured flushPeriod.
+   * Returns true and removes the entry from the map if it has expired, otherwise returns false.
+   */
+  public boolean removeIfExpired(List<String> labelValues)
```

Comment:
nit: what do you think of something like `areLabelValuesExpired` or `isMetricExpired`?